packages/jest-haste-map/src/index.js

```diff
@@ -686,7 +688,13 @@ class HasteMap extends EventEmitter {
    * Helpers
    */
   _ignore(filePath: Path): boolean {
-    return this._options.ignorePattern.test(filePath) ||
+    const ignorePattern = this._options.ignorePattern;
+    const ignoreMatched =
+      Object.prototype.toString.call(ignorePattern) === '[object RegExp]'
```

Comment:
Yep that worked actually. Will submit another PR to make that change. Thanks for the tip Simen.